### ISSUE TITLE
Bad request failure for sending messages to some contributors on Signal

### ISSUE THREAD

**Issue by phoeinx** [tactilenews/100eyes#1386]
See Sentry Issue 100EYES-1N.

tl;dr: Sometimes we receive a 400 Bad Request for some but not all contributors when sending a message.
This only affects production instances and we were not able to reproduce it yet so far. The next step is improving logging for failed message sending so that we get provided more information via Sentry the next time this happens.

Log:

The contributor with the same error on staging sadly was deleted already so that reproducing this on staging is not possible.

**Re: Rate limiting:** I had a look at the contributors which are affected by this issue. It always affects the same contributors, with an increasing number of contributors. 
As the non-receiving contributors are completely consistent and no new signal contributors joined between 05.05. and 03.06, I feel like this is probably not a rate-limiting issue.

**Re: Blocking:** I then tested a few possible ways to block us on staging but non produced this error.
Unverifying / Not verifying our number: No error.
Blocking: No error - contributor doesn’t receive message but for us it looks like a successful send. On a positive note - unblocking works perfectly well.
Archiving: No error.

**Re: Signal-Cli Settings:** I then dived into the stored data in the signal-cli-config and tried to find any patterns for the three non-receiving contributors. However, no attention calling difference was consistent for all contributors. E.g. only one contributor doesn’t have unidentifiedAccessMode enabled. The trust level for the affected Contributors varies from 0 to 2, (as for all other contributors).

**Re: Spam:** I then noticed that two out of the non-receiving contributors have not answered any requests at all. However, the third contributor, had answered (nearly) all requests before becoming non-receiving. And one other signal contributor 7 has also never answered and still receives message without problem.
I was wondering if a contributor not answering many messages sent from a server ip address would at some point be detected as Spam. (Signal currently tries to crack down on Spam.) The inconsistent behaviour here seems to indicate that this is not the case.

**Comment by mattwr18** [comment]
When would we consider this issue to be closed @phoeinx? We have added the additional logging in  [Add context to Signal errors #1391 ](https://github.com/tactilenews/100eyes/pull/1391) and while it's been just two months since we deployed this, what we've discovered so far is that the bad request can happen for a variety of reasons. Ones I have personally seen are:

- The contributor does not have their number registered with Signal any longer - support issue
- The signal server phone number did not have a name added to its profile

We could maybe improve the handling of these errors and instead of sending everything to the #monitoring channel, we could parse the error response and check for some substring that lets us know that it's a support rather than a technical issue, but I guess with the lack of diversity in the non-successful responses, we will need to handle these errors on a case-by-case basis.
